apps/portal/src/app/graphql/graphql.service.ts

```diff
@@ -15,8 +17,38 @@ export class GraphqlService {
     private authService: MsalService,
   ) {}
 
-  private getToken(): string | null {
-    return sessionStorage.getItem('accessToken');
+  private getToken(): Observable<string> {
+    const account =
+      this.authService.instance.getActiveAccount() || this.authService.instance.getAllAccounts()[0];
+
+    if (!account) {
+      // eslint-disable-next-line prefer-promise-reject-errors
+      return throwError('No active account found');
+    }
+
+    const request = {
+      scopes: [environment.apiScope],
+      account,
+    };
+    const token = this.authService.acquireTokenSilent(request).pipe(
+      switchMap((response: AuthenticationResult) => {
+        if (response && response.accessToken) {
+          return [response.accessToken];
+        }
+
+        return throwError('Failed to acquire token');
+      }),
+      catchError((error) => {
+        if (error instanceof InteractionRequiredAuthError) {
+          // fallback to interaction when silent call fails
+          this.authService.acquireTokenRedirect(request);
+          return throwError('Redirecting for token acquisition');
```

Comment:
Are we throwing this to just show message on UI?
If so can remove it.

Reply:
removed the error throwing line